src/firefly/java/edu/caltech/ipac/firefly/server/query/mos/QueryMOS.java

```diff
@@ -172,6 +188,18 @@ private URL createURL(MOSRequest req, boolean forGator) throws EndUserException,
         return new URL(url);
     }
 
+    private static String getObsDate(String obsDateParam) {
+        SimpleDateFormat receivedFormat = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ssX"); // sent by client, also ensures parsing as UTC
+        SimpleDateFormat desiredFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss"); // expected by MOST
+        try {
+            Date date = receivedFormat.parse(obsDateParam.trim());
+            return desiredFormat.format(date);
+        }
```

Comment:
@wmiipac when trying to fix this for UTC format, you broke date string without time. e.g.
```
Date date = r.parse("2023-07-17");
java.text.ParseException: Unparseable date: "2023-07-17"
	at java.base/java.text.DateFormat.parse(DateFormat.java:399)
```

Also, parsing a UTC date, then converting it back to a locale dependent format may be problematic if the server's locale is not UTC.  You should fix these.

Reply:
May be not using TimeRangePanel makes more sense for Precovery/SSO searches? SHA is the first one try use it